pkg/csi/service/common/prometheus/metrics.go

```diff
@@ -65,4 +92,19 @@ var (
 		// Possible optype - "create-volume", "delete-volume", "attach-volume", "detach-volume", "expand-volume"
 		// Possible status - "pass", "fail"
 		[]string{"voltype", "optype", "status"})
+
+	// CnsControlOpsHistVec is a histogram vector metric to observe various control
+	// operations on CNS. Note that this captures the time taken by CNS into a bucket
+	// as seen by the client(CSI in this case).
+	CnsControlOpsHistVec = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Name: "cns_volume_ops_histogram",
+		Help: "Histogram vector for CNS operations.",
+		// Creating more buckets for operations that takes few seconds and less buckets
+		// for those that are taking a long time. A CNS operation taking a long time is
+		// unexpected and we don't have to be accurate(just approximation is fine).
+		Buckets: []float64{1, 2, 3, 4, 5, 7, 10, 12, 15, 18, 20, 25, 30, 60, 120, 180, 300},
```

Comment:
Is 300(5 minutes) the maximum time that an operation can take on CNS. 
Shouldn't the maximum be 30 minutes(or whatever is the maximum timeout) for CNS tasks on VC ?

Reply:
CNS can take any time. 5 mins itself is too long which indicates problems and should be looked into with high priority.